no method error in grants transform

```
         1: from /opt/app/vitro/.rvm/gems/ruby-2.5.1@rialto/gems/traject-3.0.0/lib/traject/indexer/step.rb:138:in `block in execute'
lib/rialto/etl/configs/stanford_grants_to_sparql_statements.rb:35:in `block (2 levels) in load_config_file': undefined method `[]' for nil:NilClass (NoMethodError)
```

```
2018-10-30T09:01:37-07:00 [ERROR] 9926 : Unexpected error on record <record #12 (../rialto-sample-data/r
aw/grants/grouped_by_sunetid/mbgoodmn.json #12)>
    while executing (to_field "http://purl.org/cerif/frapo/hasStartDate" at lib/rialto/etl/configs/stanf
ord_grants_to_sparql_statements.rb:34)

    Record: {"spoNumber"=>"133334", "projectTitle"=>"Developing nanoparticle optical reporters of compre
ssive, tensile, and shear forces for use in living cells and tissues.", "piEmployeeId"=>"09726787", "piS
unetId"=>"mbgoodmn", "piFullName"=>"Goodman, Miriam B", "piAffiliationType"=>"Faculty", "piDepartmentNam
e"=>"Molecular and Cellular Phys", "projectStatus"=>"Awarded", "spoProjectDeptName"=>"Molecular and Cell
ular Phys", "spoProjectSchool"=>"School of Medicine", "directSponsorName"=>"National Institutes of Healt
h", "directSponsorForeignFlag"=>"N", "primarySponsorName"=>nil, "primarySponsorForeignFlag"=>nil, "inter
nationalFlag"=>"N", "subawardsIncludedFlag"=>"N", "agreementNumber"=>"1R21GM12987901", "agreementType"=>
"Grant", "projectStartDate"=>nil, "projectEndDate"=>nil, "edwCreateDateTime"=>"2018-09-13T03:12:35.000-0
7:00", "edwUpdateDateTime"=>"2018-09-13T03:12:35.000-07:00"}
    Exception: NoMethodError: undefined method `[]' for nil:NilClass
    lib/rialto/etl/configs/stanford_grants_to_sparql_statements.rb:35:in `block (2 levels) in load_confi
g_file'
```

--- a/lib/rialto/etl/configs/stanford_grants_to_sparql_statements.rb
+++ b/lib/rialto/etl/configs/stanford_grants_to_sparql_statements.rb
@@ -20,7 +20,6 @@ end
 # The named graph to place these triples into.
 to_field '@graph', literal(STANFORD_GRANTS_GRAPH.to_s), single: true
 to_field '@type', literal(VIVO.Grant), single: true
-
 to_field '@id', lambda { |json, accum|
   accum << RIALTO_GRANTS[json['spoNumber']]
 }, single: true
@@ -32,11 +31,13 @@ to_field RDF::Vocab::SKOS.prefLabel.to_s, extract_json('$.projectTitle'), single
 
 to_field '!' + FRAPO.hasStartDate.to_s, literal(true)
 to_field FRAPO.hasStartDate.to_s, lambda { |json, accum|
-  accum << RDF::Literal::Date.new(JsonPath.on(json, '$.projectStartDate').first[0..9])
+  maybe_date = JsonPath.on(json, '$.projectStartDate').first
+  accum << RDF::Literal::Date.new(maybe_date[0..9]) if maybe_date
 }, single: true
 to_field '!' + FRAPO.hasEndDate.to_s, literal(true)
 to_field FRAPO.hasEndDate.to_s, lambda { |json, accum|
-  accum << RDF::Literal::Date.new(JsonPath.on(json, '$.projectEndDate').first[0..9])
+  maybe_date = JsonPath.on(json, '$.projectEndDate').first
+  accum << RDF::Literal::Date.new(maybe_date[0..9]) if maybe_date
 }, single: true
 
 to_field '!' + VIVO.assignedBy, literal(true)

--- a/spec/configs/stanford_grants_to_sparql_statements_spec.rb
+++ b/spec/configs/stanford_grants_to_sparql_statements_spec.rb
@@ -244,6 +244,25 @@ RSpec.describe 'stanford_grants_to_sparql_statements' do
     end
   end
 
+  context 'when the dates are null' do
+    let(:json) do
+      <<~JSON
+        {
+          "spoNumber": "12345",
+          "projectTitle": "The Effects of Commas on Sentences",
+          "projectStartDate": null,
+          "projectEndDate": null,
+          "directSponsorName": "The William and Flora Hewlett Foundation",
+          "piEmployeeId": "12345678"
+        }
+      JSON
+    end
+
+    it 'transforms the record without errors' do
+      expect { transform(json) }.not_to raise_error
+    end
+  end
+
   describe 'add and delete grants' do
     before do
       transform(STANFORD_GRANTS_INSERT1)

